modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/ActionsComponentPropsTransformer.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import BridgeComponent from './utils/BridgeComponent';
+import Sidebar from './components/Sidebar';
+
+export default function propsTransformer({
+	items,
+	sidebarContainerSelector,
+	...otherProps
+}) {
+	const actions = {
+		showInfo() {
+			showSidebar();
+		}
+	}
+
+	const hideSidebar = () => {
+		sidebarRef.setState({open: false})
+	};
+
+	const showSidebar = () => {
+		sidebarRef.setState({open: true})
+	};
+
+	const sidebarRef = ReactDOM.render(
```

Comment:
I might be mistaken, but this should be failing linting. If you run `yarn check` it should tell you to use another `render` method, which in turn, should give you the IconContextProvider with the spritemap.

Reply:
`yarn checkFormat` (`yarn check` is [something else](https://classic.yarnpkg.com/en/docs/cli/check/)).

The lint rule in question is [`no-react-dom-render`](https://github.com/liferay/eslint-config-liferay/blob/master/plugins/eslint-plugin-liferay-portal/docs/rules/no-react-dom-render.md).

Reply:
In any case, you [should not rely on the return value of `ReactDOM.render()`](https://reactjs.org/docs/react-dom.html#render):

> `ReactDOM.render()` currently returns a reference to the root `ReactComponent` instance. However, using this return value is legacy and should be avoided because future versions of React may render components asynchronously in some cases. If you need a reference to the root `ReactComponent` instance, the preferred solution is to attach a callback ref to the root element.

Reply:
>The lint rule in question is no-react-dom-render
How do I render it then? xD

Reply:
` 40:21  error  Direct use of ReactDOM.render is discouraged; instead, use the <react:component /> JSP taglib, or do: import {render} from 'frontend-js-react-web';  liferay-portal/no-react-dom-render`

thanks `yarn checkFormat`

Reply:
Yep. The doc also says as much. You get a Clay spritemap via context for free, so it's really an amazing deal.

Reply:
![image](https://media.giphy.com/media/14sy6VGAd4BdKM/giphy.gif)

Reply:
a happy @carloslancha is a solved conversation